From 4e9728a3b0d39f64a563cc4d873afd02ec76f71e Mon Sep 17 00:00:00 2001
From: Andy Duan <fugang.duan@nxp.com>
Date: Thu, 1 Dec 2016 14:31:46 +0800
Subject: [PATCH 1304/5242] MLK-13554-02 dmaengine: fsl-edma: enable the
 correct the DMAMUX

commit  df15ec3d516df677069104970af579805c00e200 from
https://source.codeaurora.org/external/imx/linux-imx.git

- correct DMAMUX enable bit.
- correct mapping the DMAMUX source and slot.

Signed-off-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/fsl-edma.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/dma/fsl-edma.c b/drivers/dma/fsl-edma.c
index f6a5e5d..67bafba 100644
--- a/drivers/dma/fsl-edma.c
+++ b/drivers/dma/fsl-edma.c
@@ -208,7 +208,7 @@ void mux_configure32(struct fsl_edma_chan *fsl_chan, void __iomem *muxaddr,
     u32 val;
 
     if (enable)
-        val = EDMAMUX_CHCFG_ENBL << 16 | slot;
+        val = EDMAMUX_CHCFG_ENBL << 24 | slot;
     else
         val = EDMAMUX_CHCFG_DIS;
 
@@ -1054,7 +1054,9 @@ static int fsl_edma_probe(struct platform_device *pdev)
 		vchan_init(&fsl_chan->vchan, &fsl_edma->dma_dev);
 
 		edma_writew(fsl_edma, 0x0, fsl_edma->membase + EDMA_TCD_CSR(i));
+		fsl_chan->vchan.chan.chan_id = i;
 		fsl_edma_chan_mux(fsl_chan, 0, false);
+		fsl_chan->vchan.chan.chan_id = 0;
 	}
 
 	edma_writel(fsl_edma, ~0, fsl_edma->membase + EDMA_INTR);
-- 
1.7.9.5

